examples/gwlb/main.tf

```diff
@@ -0,0 +1,142 @@
+resource "azurerm_resource_group" "this" {
+  name     = "${var.name_prefix}${var.resource_group_name}"
+  location = var.location
+}
+
+module "vnet" {
+  for_each = var.vnets
+  source   = "../../modules/vnet"
+
+  virtual_network_name = "${var.name_prefix}${each.value.name}"
+  location             = var.location
+  resource_group_name  = azurerm_resource_group.this.name
+  address_space        = each.value.address_space
+
+  network_security_groups = each.value.network_security_groups
+  subnets                 = each.value.subnets
+
+  tags = var.tags
+}
+
+module "gwlb" {
+  source = "../../modules/gwlb"
+
+  name                = "${var.name_prefix}${var.gwlb.name}"
+  location            = var.location
+  resource_group_name = azurerm_resource_group.this.name
+
+  backends = var.gwlb.backends
+
+  frontend_ip_config = {
+    subnet_id = module.vnet[var.gwlb.vnet_key].subnet_ids[var.gwlb.subnet_key]
+  }
+
+  health_probe = {
+    port = var.gwlb.health_probe.port
+  }
+}
+
+module "vmseries" {
+  for_each = var.vmseries
+  source   = "../../modules/vmseries"
+
+  name                = "${var.name_prefix}${each.key}"
+  location            = var.location
+  resource_group_name = azurerm_resource_group.this.name
+  avzone              = each.value.avzone
+  username            = var.vmseries_common.username
+  ssh_keys            = var.vmseries_common.ssh_keys
+  img_version         = var.vmseries_common.img_version
+  img_sku             = var.vmseries_common.img_sku
+  vm_size             = var.vmseries_common.vm_size
+  bootstrap_options   = var.vmseries_common.bootstrap_options
+
+  interfaces = [for interface in var.vmseries_common.interfaces :
+    {
+      name                = "${var.name_prefix}${each.key}-${interface.name}"
+      subnet_id           = module.vnet[var.vmseries_common.vnet_key].subnet_ids[interface.name]
+      create_public_ip    = try(interface.create_public_ip, false)
+      enable_backend_pool = try(interface.enable_backend_pool, false)
+      lb_backend_pool_id  = try(interface.enable_backend_pool, false) ? module.gwlb.backend_pool_ids[interface.gwlb_backend_key] : null
+    }
+  ]
+
+  tags = var.tags
+}
+
+module "applb" {
+  source = "../../modules/loadbalancer"
+
+  name                = "${var.name_prefix}application"
+  location            = var.location
+  resource_group_name = azurerm_resource_group.this.name
+  frontend_ips = {
```

Comment:
This is static map, do you maybe think to put it into the *.tfvars file?

Reply:
I would say we can leave it like this. A refactor of all examples is coming. It would touch not only this map, but probably the rest of the code as well. Maybe it does not make sense to do a mix of new and all approaches as it might mean extra work at the end.